src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -90,6 +94,16 @@ public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull St
         this.iamMfaSerialNumber = Util.fixNull(iamMfaSerialNumber);
     }
 
+    @CheckForNull
+    public String getAwsRegion() {
+        return awsRegion;
+    }
+
+    @DataBoundSetter
+    public void setAwsRegion(@CheckForNull String awsRegion) {
+        this.awsRegion = Util.fixNull(awsRegion);
```

Comment:
If `null` or empty it should either use the default value or raise an error. Be aware that being a `@DataBoundSetter` implies that it is going to be called by stapler after using the validator below, but it can also be called directly so the same validation should be in place for programmatic use.

Reply:
@andresrc I'm not sure if what I did is the correct way, also because I'm not too familiar with the best practices for this plugin. Please let me know if that is correct or if not what do I need to change. thanks again for your time for reviewing this and providing feedback, it is very useful to learn